src/core/stokesian_dynamics/CMakeLists.txt

```
@@ -0,0 +1,17 @@
+add_library(sd_interface SHARED sd_interface.cpp)
+target_include_directories(sd_interface PRIVATE
+  ${CMAKE_BINARY_DIR}/src/core
+  ${CMAKE_SOURCE_DIR}/src/core
+  ${CMAKE_SOURCE_DIR}/src/core/stokesian_dynamics
+  ${CMAKE_SOURCE_DIR}/libs/stokesian_dynamics
+)
+target_link_libraries(sd_interface PUBLIC EspressoConfig Boost::mpi MPI::MPI_CXX Random123 utils)
```

shouldn't `EspressoConfig` be private?
```suggestion
target_link_libraries(sd_interface PRIVATE EspressoConfig PUBLIC Boost::mpi MPI::MPI_CXX Random123 utils)
```

---

I guess you're right. I don't recall any of the variables provided by `EspressoConfig` being used in SD.

---

In a nutshell, the `EspressoConfig` library allows us to write `#include "config.hpp"` in the source code of the `sd_interface` library, and the `PRIVATE` option will prevent compiler flags from `EspressoConfig` to propagate into libraries that include `sd_interface`, e.g. `EspressoCore`.